account_invoice_start_end_dates_tax_change/wizards/account_move_apply_tax_change.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+
+# NOTE: method copied from recent Odoo versions (odoo.tools.date_utils)
+def date_subtract(value, *args, **kwargs):
+    """
+    Return the difference between ``value`` and a :class:`relativedelta`.
+
+    :param value: initial date or datetime.
+    :param args: positional args to pass directly to :class:`relativedelta`.
+    :param kwargs: keyword args to pass directly to :class:`relativedelta`.
+    :return: the resulting date/datetime.
+    """
+    return value - relativedelta(*args, **kwargs)
+
+
+class AccountMoveApplyTaxChange(models.TransientModel):
+    _inherit = "account.move.apply.tax.change"
+
+    def _skip_line(self, line):
+        if line.start_date or line.end_date:
+            if line.start_date <= self.tax_change_id.date < line.end_date:
+                return False
+            return True
+        return super(AccountMoveApplyTaxChange, self)._skip_line(line)
+
+    def _change_taxes(self, line):
+        if line.start_date or line.end_date:
+            # Start & end dates defined in the scope of the tax changes
+            if line.start_date <= self.tax_change_id.date < line.end_date:
+                # Split the invoice line by applying a prorata on the unit price
+                # NOTE: if the tax change occurs on 2024-02-01, the end date
+                # of the split invoice line will be the day before (2024-01-31)
+                # such as it doesn't overlap with the start date of the new line.
+                tax_change_date = fields.Date.from_string(self.tax_change_id.date)
+                new_end_date = date_subtract(tax_change_date, days=1)
+                # Dates subtractions are not inclusives so we add +1 day each time
+                end_date = fields.Date.from_string(line.end_date)
+                start_date = fields.Date.from_string(line.start_date)
+                total_days = (end_date - start_date).days + 1
+                days_until_change = (new_end_date - start_date).days + 1
+                days_from_change = (end_date - tax_change_date).days + 1
+                price_unit1 = line.price_unit * days_until_change / total_days
+                price_unit2 = line.price_unit * days_from_change / total_days
```

Comment:
```suggestion
                price_unit2 = line.price_unit - price_unit1
```
:warning: we have to make sure the sum of both new price units equals the original one, which could not be the case with precision roundings.